### ISSUE TITLE
Can't upgrade salt-minion in Amazon Linux AMI release 2018.03 via yum

### ISSUE THREAD

**Issue by gabrielperezs** [saltstack/salt#47593]
### Description of Issue/Question
After upgrade via yum with the offical repos of saltstack the salt-minion don't work with the services. 

The issue is related exactly with this code https://github.com/saltstack/salt/commit/e4d1d5bf110f4e83b7c26f8e6b46c4cb99e00652#diff-6b90f8759e00cc326a98173a1685d3a1 what is not included in the RPM package https://repo.saltstack.com/yum/amazon/latest/x86_64/latest/salt-2018.3.0-1.amzn1.noarch.rpm

### Setup
```yaml
smart-relayer:
  service.running:
    - enable: True
    - restart: True
    - watch:
      - file: /usr/local/bin/smart-relayer
```
### Steps to Reproduce Issue
This bug was reported here #47271 and here #47561
```
salt-call state.sls smart-relayer test=True
```
```
----------
          ID: smart-relayer
    Function: service.running
      Result: False
     Comment: State 'service.running' was not found in SLS 'smart-relayer'
              Reason: 'service' __virtual__ returned False: No service execution module loaded: check support for service management on Amazon Linux AMI-2018
     Changes:   
----------
```

### Versions Report
```
Salt Version:
           Salt: 2018.3.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.14 (default, May  2 2018, 18:31:34)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist:   
         locale: UTF-8
        machine: x86_64
        release: 4.9.20-11.31.amzn1.x86_64
         system: Linux
        version: Not Installed
```

**Comment by gabrielperezs** [comment]
@Ch3LL Yes. For example, just commenting those lines:

```python
        if __grains__['os'] in ('RedHat', 'CentOS', 'ScientificLinux', 'OEL', 'CloudLinux'):
            if osrelease_major >= 7:
                return (
                    False,
                    'RedHat-based distros >= version 7 use systemd, will not '
                    'load rh_service.py as virtual \'service\''
                )
        # if __grains__['os'] == 'Amazon':
        #     if int(osrelease_major) in (2016, 2017):
        #         return __virtualname__
        #     else:
        #         return (
        #             False,
        #             'Amazon Linux >= version 2 uses systemd. Will not '
        #             'load rh_service.py as virtual \'service\''
        #         )
        return __virtualname__
```

The salt-minion is able to run a service:
```
----------
          ID: smart-relayer
    Function: service.running
      Result: True
     Comment: The service smart-relayer is already running
     Started: 17:08:28.910827
    Duration: 6.49 ms
     Changes:   
----------
```

**Comment by Ch3LL** [comment]
okay so since this has been resolved by that PR we can go ahead and close this issue. That fix will be in 2017.7.6 release which we are currently working on.
